Story/StoryHooks.cs

```diff
@@ -802,6 +809,95 @@ private void RainWorldGame_GoToDeathScreen(On.RainWorldGame.orig_GoToDeathScreen
             orig(self);
         }
 
+        private void SleepAndDeathScreen_AddPassageButton(On.Menu.SleepAndDeathScreen.orig_AddPassageButton orig, Menu.SleepAndDeathScreen self, bool buttonBlack)
+        {
+            if (isStoryMode(out _) && !OnlineManager.lobby.isOwner) return;
+            orig(self, buttonBlack);
+        }
+
+        private void CustomEndGameScreen_GetDataFromSleepScreen(On.Menu.CustomEndGameScreen.orig_GetDataFromSleepScreen orig, Menu.CustomEndGameScreen self, WinState.EndgameID endGameID)
+        {
+            if (isStoryMode(out _) && OnlineManager.lobby.isOwner)
+            {
+                foreach (var player in OnlineManager.players)
+                {
+                    if (!player.isMe) player.InvokeOnceRPC(RPCs.GoToPassageScreen, endGameID);
+                }
+            }
+
+            orig(self, endGameID);
+        }
+
+        private void FastTravelScreen_ctor(On.Menu.FastTravelScreen.orig_ctor orig, Menu.FastTravelScreen self, ProcessManager manager, ProcessManager.ProcessID ID)
+        {
+            orig(self, manager, ID);
+            if (isStoryMode(out _) && !OnlineManager.lobby.isOwner)
+            {
+                StoryMenuHelpers.RemoveMenuObjects(self.mapButtonPrompt, self.prevButton, self.nextButton, self.chooseButton, self.buttonInstruction);
+            }
+        }
+
+        private void FastTravelScreen_ctor_ClientDontFilterRegions(ILContext il)
+        {
+            try
```

Comment:
Nitpick: commenting the changes to the IL. I can infer from the method name though. Your call.

Reply:
it's just
```diff
  bool flag = false;
+ if (isStoryMode(out _) && !OnlineManager.lobby.isOwner) flag = true;
```

and to understand what flag does you'll have to look at the IL anyway